manageiq-operator/README.md

```diff
@@ -2,68 +2,125 @@
 
-If you want to use a custom TLS certificate, it can be created with `oc create secret tls tls-secret --cert=tls.crt --key=tls.key` and setting the secret name as `tlsSecret` in `manageiq.org_v1alpha1_manageiq_cr.yaml`.
+## Running ManageIQ under operator control.
 
-## Run Operator
+There are five high level steps for running ManageIQ under operator control:
 
-Deploy the ManageIQ CRD
+  + Step 1. Set up RBAC
+  + Step 2. Deploy the ManageIQ Custom Resource Definition (CRD) (If it is not already running)
+  + Step 3. Run The Operator
+  + Step 4. Perform any optional custom configurations
+  + Step 5. Run ManageIQ by creating the Custom Resource (CR)
 
-```bash
-$ oc create -f deploy/crds/manageiq.org_manageiqs_crd.yaml
-```
+The details of the five steps are as follows:
 
-### Option A: Run Locally(outside of cluster)
-
-In project root directory, run
+### Step 1. Set up RBAC
 
 ```bash 
-$ operator-sdk run --local --namespace=<yournamespace>
+$ oc create -f deploy/role.yaml
+$ oc create -f deploy/role_binding.yaml
+$ oc create -f deploy/service_account.yaml
 ```
 
-### Option B: Run Inside Cluster
+### Step 2. Deploy the ManageIQ Custom Resource Definition (CRD)
 
-1. Build the operator image:
+The ManageIQ CRD only needs to be running once on the cluster. If it is already running it does not need to be restarted.
+
+To determine if it is already available execute command:
 
 ```bash
-$ operator-sdk build docker.io/example/manageiq-operator:latest
+$ oc get crds | grep manageiqs.manageiq.org
 ```
 
-2. Update the operator deployment with the new image:
+If it is not already available it can be deployed with command:
 
 ```bash
-$ sed -i 's|docker.io/manageiq/manageiq-operator:latest|docker.io/example/manageiq-operator:latest|g' deploy/operator.yaml
+$ oc create -f deploy/crds/manageiq.org_manageiqs_crd.yaml
 ```
 
-3. Push the new image to the registry:
+### Step 3. Run The Operator
+
+There are three different ways the operator can be run.
+
+
++ #### Option 1: Run the latest ManageIQ image from the registry
+
+  The default in the operator.yaml is for the latest manageiq-operator image.
+  So no change is required. Simply create the operator.
+
+  ```bash
+  $ oc create -f deploy/operator.yaml
+  ```
+
++ #### Option 2: Run your own custom ManageIQ image inside the Cluster
+
+  1 - Build your operator image:
+
+    ```bash
+    $ operator-sdk build docker.io/<your location>/manageiq-operator:latest
+    ```
+
+  2 - Update the operator deployment yaml file with your custome image:
+
+    ```bash
+    $ sed -i 's|docker.io/manageiq/manageiq-operator:latest|docker.io/<your location>/manageiq-operator:latest|g' deploy/operator.yaml
+    ```
+
+  3 - Push your new custom image to the registry:
```

Comment:
This should probably be after Step 1 but before 2 since it's closely related to Step 1